fw/ctrl/Makefile

```diff
@@ -48,9 +42,9 @@ BUILD_T=$$(date +%Y-%m-%d)
 BUILD_TIME=\"$(BUILD_T)\"
 BUILD_N=$$(cat build_num.txt)
 BUILD_NUM=\"$(BUILD_N)\"
-BUILD_R=$$(git rev-parse --verify HEAD)
+BUILD_R=$$(git rev-parse --verify HEAD 2>/dev/null || printf '%s\n' "norev")
 BUILD_REV=\"$(BUILD_R)\"
-BUILD_TG=$$(git describe --tags --abbrev=0)
+BUILD_TG=$$(git describe --tags --abbrev=0 2>/dev/null || printf '%s\n' "notag")
```

Comment:
Avoids errors when rebuilding the source tree copied to a system without git being available, or when building from a shallow clone.